src/sprite_resource.c

```diff
@@ -0,0 +1,884 @@
+static void SpriteResourceCollection_InitResFromNARC(SpriteResourceCollection *spriteResources, SpriteResource *spriteRes, NARC *narc, int memberIdx, BOOL compressed, int id, NNS_G2D_VRAM_TYPE vramType, int paletteIdx, enum SpriteResourceType type, enum HeapId heapID, u32 param10);
+static void *SpriteUtil_ReadNARCMember(NARC *narc, u32 memberIdx, BOOL compressed, u32 heapID, u32 param4);
+static void SpriteResource_UnpackData(SpriteResource *spriteRes, enum SpriteResourceType type, NNS_G2D_VRAM_TYPE vramType, int paletteIdx, enum HeapId heapID);
+static TileResourceData *SpriteUtil_UnpackTileResource(void *rawData, NNS_G2D_VRAM_TYPE vramType, enum HeapId heapID);
+static PaletteResourceData *SpriteUtil_UnpackPaletteResource(void *rawData, NNS_G2D_VRAM_TYPE vramType, int paletteIdx, enum HeapId heapID);
+static SpriteResourceData *SpriteUtil_UnpackSpriteResource(void *rawData, enum HeapId heapID);
+static SpriteAnimResourceData *SpriteUtil_UnpackSpriteAnimResource(void *rawData, enum HeapId heapID);
+static MultiSpriteResourceData *SpriteUtil_UnpackMultiSpriteResource(void *rawData, enum HeapId heapID);
+static MultiSpriteAnimResourceData *SpriteUtil_UnpackMultiSpriteAnimResource(void *rawData, enum HeapId heapID);
+static void *SpriteResource_GetData(const SpriteResource *spriteRes);
+static void SpriteResource_FreeData(SpriteResource *spriteRes);
+static int SpriteResourceTableEntryNARC_GetEntryCount(const SpriteResourceTableEntryNARC *entries);
+
+SpriteResourceCollection *SpriteResourceCollection_New(int capacity, enum SpriteResourceType type, enum HeapId heapID)
+{
+    SpriteResourceCollection *spriteResources = Heap_AllocFromHeap(heapID, sizeof(SpriteResourceCollection));
+    spriteResources->collection = ResourceCollection_New(capacity, heapID);
+    spriteResources->resources = Heap_AllocFromHeap(heapID, sizeof(SpriteResource) * capacity);
+
+    memset(spriteResources->resources, 0, sizeof(SpriteResource) * capacity);
+
+    spriteResources->capacity = capacity;
+    spriteResources->count = 0;
+    spriteResources->type = type;
+
+    return spriteResources;
+}
+
+void SpriteResourceCollection_Delete(SpriteResourceCollection *spriteResources)
+{
+    GF_ASSERT(spriteResources);
+    GF_ASSERT(spriteResources->collection);
+    GF_ASSERT(spriteResources->resources);
+
+    SpriteResourceCollection_Clear(spriteResources);
+
+    ResourceCollection_Delete(spriteResources->collection);
+    spriteResources->collection = NULL;
+
+    Heap_FreeToHeap(spriteResources->resources);
+    spriteResources->resources = NULL;
+
+    Heap_FreeToHeap(spriteResources);
+    spriteResources = NULL;
+}
+
+SpriteResource *SpriteResourceCollection_AddFromTable(SpriteResourceCollection *spriteResources,
+    const SpriteResourceTable *table,
+    int index,
+    enum HeapId heapID)
```

Comment:
**question:** Did `clang-format` mandate this?

Reply:
It's a result of running clang-format, yes, but I already had some weird formatting there to begin with so idk if it just tried to respect that. Because if I put it all on one line and then run the formatter again it doesn't do this. Want me to change it?

Reply:
Yes, preferably. Function definitions should adhere to a single line, ideally.